dsl/src/test/java/org/mule/test/extension/dsl/ArtifactDeclarationSerializerTestCase.java

```diff
@@ -582,38 +618,58 @@ private static Object createNoMuleComponentsArtifactDeclaration() {
             .withComponent(jms.newOperation("publish")
                 .withConfig("config")
                 .withParameterGroup(newParameterGroup()
-                    .withParameter("destination", "#[initialDestination]")
+                    .withParameter("destination", createStringParameter("#[initialDestination]"))
                     .getDeclaration())
                 .withParameterGroup(newParameterGroup("Message")
-                    .withParameter("body", "#[payload]")
-                    .withParameter("properties", "#[{(initialProperty): propertyValue}]")
+                    .withParameter("body", createStringParameter("#[payload]"))
+                    .withParameter("properties", createStringParameter("#[{(initialProperty): propertyValue}]"))
                     .getDeclaration())
                 .getDeclaration())
             .withComponent(http.newOperation("request")
                 .withConfig("httpRequester")
                 .withParameterGroup(newParameterGroup("URI Settings")
-                    .withParameter("path", "/nested")
+                    .withParameter("path", createStringParameter("/nested"))
                     .getDeclaration())
                 .withParameterGroup(newParameterGroup()
-                    .withParameter("method", "POST")
+                    .withParameter("method", createStringParameter("POST"))
```

Comment:
this is not actually necessary, right? the user can still do `.withParameter("method", "POST")`